### ISSUE TITLE
Bugs in azureiotedge-diagnostics:1.0.10

### ISSUE THREAD

**Issue by sacvalleytech** [Azure/iotedge#3730]
## Expected Behavior
Running `iotedge check` should not throw errors when using a VALID configuration

## Current Behavior

`iotedge check` fails with:

config.yaml has correct URIs for daemon mgmt endpoint - Error
    One or more errors occurred. (Response status code does not indicate success: 404 (Not Found).)

## Steps to Reproduce
Provide a detailed set of steps to reproduce the bug.
1. edit `config.yaml` and set:

```
connect: 
    management_uri: "http://172.18.0.2:15581"

```

2.  run `iotedge check`
3. ERROR: 

```
 config.yaml has correct URIs for daemon mgmt endpoint - Error
    One or more errors occurred. (Response status code does not indicate success: 404 (Not Found).)
```

### Device Information
* Host OS: Windows 10 (WSL2 Ubuntu 18.04)
* Architecture: amd64
* Container OS: Linux

### Runtime Versions
* iotedged: 1.0.10
* Edge Agent: 1.0.10 
* Edge Hub: 1.0.10
* Docker/Moby: 19.03.13 

## Additional Information

There are multiple bugs... however since they are all closely related I opted to file under a single issue...

**BUG - 1**

Inspecting the Docker container that is created for the test: `mcr.microsoft.com/azureiotedge-diagnostics:1.0.10` shows the management URL has a trailing `/`... this is causing the 404 (my management URL in the config.yaml did not end with a `/` so this is being appended automatically

```
            "Cmd": [
                "dotnet",
                "IotedgeDiagnosticsDotnet.dll",
                "edge-agent",
                "--management-uri",
                "http://172.18.0.2:15581/"
            ],

```

This can be proven by running the following command: `docker run --network=azure-iot-edge -it mcr.microsoft.com/azureiotedge-diagnostics:1.0.10 dotnet IotedgeDiagnosticsDotnet.dll edge-agent --management-uri http://172.18.0.2:15581/`

Returns: One or more errors occurred. (Response status code does not indicate success: 404 (Not Found).)

vs.

`docker run --network=azure-iot-edge -it mcr.microsoft.com/azureiotedge-diagnostics:1.0.10 dotnet IotedgeDiagnosticsDotnet.dll edge-agent --management-uri http://172.18.0.2:15581`

One or more errors occurred. (Got bad response: {"modules: [...]

**BUG - 2**

Notice how removing the `/` now returns a non 404 error... but there is still an error message in the response. However the error message contains the full response from the API containing a JSON payload with all the modules listed.

It looks like the culprit is this source code:

https://github.com/Azure/iotedge/blob/237f9227b2b49861fb19fa7ca246c57784549181/edge-modules/iotedge-diagnostics-dotnet/src/Program.cs#L72-L83

1) Line 75 ensures there is a successful status code

2) Line 76: reads the responses content (i.e. body) which is a JSON payload

3) However line 80: expects the content (body) to start w/`HTTP/1.1 200 OK`... this code seems to assume the full (headers + body) response is read into the variable not just the body.

Lines 80-83 are really not needed as line 75 has already validated the status code...

4) Running `wget -S -O -  http://172.18.0.2:15581/modules/?api-version=2018-06-28`  shows the correct response...

```
HTTP/1.1 200 OK
  content-type: application/json
  content-length: 13300
  date: Fri, 16 Oct 2020 04:12:04 GMT

writing to stdout
{"modules":[{"id":"id","

```

**Comment by massand** [comment]
Thanks for reporting this. Our team is taking a look.

**Comment by FMauroy** [comment]
On a side note, on Windows, we have this error at the same location:

`Only 'http' and 'https' schemes are allowed. (Parameter 'requestUri')`

**Comment by lfitchett** [comment]
This is fixed by #3950 and is in the current 1.0.10.2 release.

**Comment by sacvalleytech** [comment]
@lfitchett /facepalm... seriously? not even close to fixed... I suggest you read the original bug report again.
